File: hotel-reservation-api/app/Models/RoomCategory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class RoomCategory extends Model
{
    protected $fillable = [
        'name',
        'description',
        'base_price',
        'max_guests',
        'amenities',
        'image_url',  // Single image per category
        'is_active',
    ];

    protected $casts = [
        'base_price' => 'decimal:2',
        'amenities' => 'array',
        'is_active' => 'boolean',
    ];

    protected $appends = ['image_full_url'];

    public function rooms()
    {
        return $this->hasMany(Room::class, 'category_id');
    }

    public function availableRooms()
    {
        return $this->rooms()->where('status', 'available');
    }

    public function getImageFullUrlAttribute()
    {
        if ($this->image_url) {
            // If starts with http/https, return as is
            if (str_starts_with($this->image_url, 'http')) {
                return $this->image_url;
            }
            // Otherwise, prepend storage URL
            return asset('storage/' . $this->image_url);
        }
        return asset('images/placeholder-room.jpg');
    }
}

File: hotel_reservation/lib/models/room_category.dart
class RoomCategory {
  final int id;
  final String name;
  final String? description;
  final double basePrice;
  final int maxGuests;
  final List<String> amenities;
  final String? imageUrl;  // Single image per category
  final bool isActive;

  RoomCategory({
    required this.id,
    required this.name,
    this.description,
    required this.basePrice,
    required this.maxGuests,
    required this.amenities,
    this.imageUrl,
    required this.isActive,
  });

  factory RoomCategory.fromJson(Map<String, dynamic> json) {
    return RoomCategory(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      basePrice: double.parse(json['base_price'].toString()),
      maxGuests: json['max_guests'],
      amenities: List<String>.from(json['amenities'] ?? []),
      imageUrl: json['image_url'] ?? json['image_full_url'],
      isActive: json['is_active'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'base_price': basePrice,
      'max_guests': maxGuests,
      'amenities': amenities,
      'image_url': imageUrl,
      'is_active': isActive,
    };
  }
}